src/utilities/do-try.ts

```diff
@@ -132,6 +165,13 @@ class DoSync<TResourceType, TReturnVal = void> {
         try {
             return this.workload();
         } catch (e) {
+            if (e instanceof ResultRecord) {
+                DoSync.config.defaultErrorHandler?.(e, undefined);
+                this.catchHandler?.(e);
+                return;
+            }
+
+            DoSync.config.defaultErrorHandler?.(undefined, e);
             this.catchHandler?.(e);

```

Comment:
Is this supposed to call `catchHandler` with `undefined, e`? I thought the first param was meant to handle if `e` was an instance of `ResultRecord`

Reply:
Take a look at the `catch()` method on `DoSync` class. It takes a singular `err?: any` and type-checks it against `ResultRecord`; then it passes the value appropriately into the handler passed into the `catch()` method. This `catchHandler` basically wraps that handler, and accepts the literal `any` value from the `catch(e) { }` block